exchanges/stream/buffer/buffer.go

```diff
@@ -57,27 +64,40 @@ func (w *Orderbook) validate(u *Update) error {
 	return nil
 }
 
-// Update updates a local buffer using bid targets and ask targets then updates
-// main orderbook
-// Volume == 0; deletion at price target
-// Price target not found; append of price target
-// Price target found; amend volume of price target
+// Update updates a stored pointer to an orderbook.Depth struct containing a
+// linked list, this switches between the usage of a buffered update
 func (w *Orderbook) Update(u *Update) error {
 	if err := w.validate(u); err != nil {
 		return err
 	}
 	w.m.Lock()
 	defer w.m.Unlock()
-	obLookup, ok := w.ob[u.Pair.Base][u.Pair.Quote][u.Asset]
+	book, ok := w.ob[u.Pair.Base][u.Pair.Quote][u.Asset]
 	if !ok {
-		return fmt.Errorf("ob.Base could not be found for Exchange %s CurrencyPair: %s AssetType: %s",
+		return fmt.Errorf("%w for Exchange %s CurrencyPair: %s AssetType: %s",
+			errDepthNotFound,
 			w.exchangeName,
 			u.Pair,
 			u.Asset)
 	}
 
+	// Checks for when the rest protocol overwrites a streaming dominated book
+	// will stop updating book via incremental updates. This occurs because our
+	// sync manager (engine/sync.go) timer has elapsed for streaming. Usually
+	// because the book is highly illiquid.
+	if book.ob.IsRestSnapshot() {
+		return fmt.Errorf("%w for Exchange %s CurrencyPair: %s AssetType: %s",
```

Comment:
I'm not expecting this in this PR, but this change seems like there is no recovery for websocket. If this ever gets hit, any websocket orderbook update that comes through will see this error as there is nothing to reset the orderbook to not have this be true. Perhaps add a TODO to signify that such an update will happen?
For a simple proof of concept, I basically mean reading the error on the {{exchange}}_websocket file to check that its `ErrRestOverwrite` and un/resubscribe. 
Or maybe just a first time `warn` the user to consider extending `synctimeoutwebsocket`